compiler/src/iree/compiler/Codegen/Dialect/GPU/TargetUtils/ConfigUtils.cpp

```diff
@@ -7,106 +7,208 @@
-  // TODO: We shouldn't need this check, however loop fusion currently requires
-  // loop trip counts to be identical, meaning we need to use a num_threads
-  // variant of tiling. Remove this and simply return the preferred vector size
-  // once loop fusion can resolve the forall properly.
-  if (llvm::any_of(loopRanges,
-                   [](int64_t s) { return ShapedType::isDynamic(s); })) {
-    return {};
+LogicalResult setMatmulLoweringConfig(IREE::GPU::TargetAttr target,
+                                      mlir::FunctionOpInterface entryPoint,
+                                      Operation *op) {
+  auto linalgOp = dyn_cast<linalg::LinalgOp>(op);
+  if (!linalgOp) {
+    return failure();
   }
 
-  int64_t flatNumTrips = std::accumulate(loopRanges.begin(), loopRanges.end(),
-                                         1, std::multiplies<int64_t>());
-  if (flatNumTrips % numThreads != 0) {
-    return {};
+  if (target.getWgp().getMma().empty())
+    return failure();
+
+  const int64_t targetSubgroupSize = target.getPreferredSubgroupSize();
+
+  SmallVector<int64_t, 4> bounds = linalgOp.getStaticLoopRanges();
+  FailureOr<mlir::linalg::ContractionDimensions> contractionDims =
+      mlir::linalg::inferContractionDims(linalgOp);
+  if (failed(contractionDims)) {
+    return failure();
   }
-  int64_t maxVectorSize = flatNumTrips / numThreads;
 
-  while (maxVectorSize % vectorSize != 0) {
-    vectorSize /= 2;
+  if (contractionDims->k.size() < 1 || contractionDims->m.size() < 1 ||
+      contractionDims->n.size() < 1) {
+    return failure();
   }
 
-  SmallVector<int64_t> tileSizes(loopRanges.size(), 0);
-  tileSizes.back() = vectorSize;
-  int64_t residualNumThreads = numThreads / (loopRanges.back() / vectorSize);
-  for (int i = tileSizes.size() - 2, e = 0; i >= e; --i) {
-    if (loopRanges[i] >= residualNumThreads) {
-      tileSizes[i] = loopRanges[i] / residualNumThreads;
-      residualNumThreads = 1;
-      break;
-    }
-    tileSizes[i] = 1;
-    residualNumThreads /= loopRanges[i];
+  // For now we are not being smart and trying to reshape dimensions to allow
+  // for better usage of intrinsics, and instead are tiling all dimensions
+  // except the inner most m, n, and k dimensions to 1.
+  int64_t mDim = contractionDims->m.back();
+  int64_t nDim = contractionDims->n.back();
+  int64_t kDim = contractionDims->k.back();
+
+  // Dynamic dims are expected to be taken care of earlier in the pipeline.
+  if (ShapedType::isDynamic(bounds[mDim]) ||
+      ShapedType::isDynamic(bounds[nDim]) ||
+      ShapedType::isDynamic(bounds[kDim])) {
+    return failure();
   }
-  return tileSizes;
-}
 
-SmallVector<int64_t> deriveLinalgOpThreadTileSizes(linalg::LinalgOp linalgOp,
-                                                   int64_t numThreads) {
-  if (!linalgOp.hasPureTensorSemantics()) {
-    return {};
+  // Bail out on matvec-like cases.
+  if (bounds[mDim] == 1 || bounds[nDim] == 1) {
+    return failure();
```

Comment:
Is this necessary? Will it be captured by the MMA heuristics?

Reply:
hmmm yeah, probably not right now. I'll drop it, but eventually we'll need to add it back.